### PR TITLE
[flutter_migrate] Fix intl version to support localized apps

### PR DESCRIPTION
**Pull request by frederikstonge**
Uses intl: any instead of hardcoded version to avoid conflicts with flutter_localizations from sdk.

for issue https://github.com/flutter/flutter/issues/130493

### CODE REVIEW

**Review by stuartmorgan — CHANGES_REQUESTED** [review]
Thanks for the contribution!

File: packages/flutter_migrate/pubspec.yaml [hunk 1]
```diff
@@ -12,7 +12,7 @@ dependencies:
   args: ^2.3.1
   convert: 3.0.2
   file: 6.1.4
-  intl: 0.17.0
+  intl: any
```
[the remark this answers is not in the document]
Comment (frederikstonge) [reply packages/flutter_migrate/pubspec.yaml:15]: > Use of `any` in a published package is an anti-pattern; it subjects clients of the package to arbitrary breakage at any point in the future. If a package has been tested with multiple major version of a dependency, the way to express that is with a version range, not by claiming compatibility with every past and future version that has or will exist.

I understand. The thing is that `flutter_migrate` is supposed to allow any apps to be updated to a newer version. So any app that will use flutter_localization from the sdk has a dependency on a different intl version. (0.18.0 with current stable).

From what I'm seeing, intl is used only in the logger to format numbers. Should we look for an alternative?

[the remark this answers is not in the document]
Comment (frederikstonge) [reply packages/flutter_migrate/pubspec.yaml:15]: image

[the remark this answers is not in the document]
Comment (stuartmorgan) [reply packages/flutter_migrate/pubspec.yaml:15]: > The thing is that `flutter_migrate` is supposed to allow any apps to be updated to a newer version. So any app that will use flutter_localization from the sdk has a dependency on a different intl version. (0.18.0 with current stable).

The Flutter version that was used to create a project being migrated and the Flutter version that someone is using to run the tool are unrelated, so I don't understand why that would matter.

[the remark this answers is not in the document]
Comment (frederikstonge) [reply packages/flutter_migrate/pubspec.yaml:15]: Oh, so it's supposed to be installed as a global package? I was adding it as a dev dependency to my project...

[the remark this answers is not in the document]
Comment (frederikstonge) [reply packages/flutter_migrate/pubspec.yaml:15]: Oh wow I get it. Sorry about the waste of time.

**Pull request closed without merging**